api/src/modules/assignor/assignor.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/database/prisma.service';
+import { AssignorDTO } from '../../dtos/assignor.dto';
+
+
+@Injectable()
+export class AssignorService {
+  constructor(private prisma: PrismaService) {}
+
+  async create(data: AssignorDTO) {
+    const assignorExists = await this.prisma.assignor.findFirst({
+      where: {
+        document: data.document,
+      },
+    });
+
+    if (assignorExists) {
+      throw new Error('Assignor already exists');
+    }
+
+    const assignor = await this.prisma.assignor.create({
+      data,
+    });
+ 
+    return assignor;
```

Comment:
aqui poderia retornar diretamente no this.prisma.assignor, sem a necessidade de criar uma variavel que não será utilizada posteriormente.